File: strStr.py
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        index = -1
        for i in range(len(haystack) - len(needle) + 1):
            if haystack[i] == needle[0]:
                index = i
                for k in range(len(needle)):
                    if haystack[k+i] != needle[k]:
                        index = -1
                        break
                
                if index != -1:
                    return index
        
        return index


s = Solution()
print(s.strStr(haystack = "a", needle = "a"))

File: golang/str_str.go
package main

import "unicode/utf8"

func strStr(haystack string, needle string) int {
	needleLen := utf8.RuneCountInString(needle)
	if needleLen > utf8.RuneCountInString(haystack) {
		return -1
	}

	for i := 0; i < utf8.RuneCountInString(haystack)-needleLen+1; i++ {
		if haystack[i] == needle[0] {
			if haystack[i:i+needleLen] == needle {
				return i
			}
		}
	}

	return -1
}